.github/workflows/dockerhub-release-matrix.yml

```diff
@@ -0,0 +1,197 @@
+          cache-to: type=gha,mode=max,scope=${{ github.ref_name }}-latest-${{ matrix.arch }}
+          file: ${{ matrix.postgres.dockerfile }}
+  merge_manifest:
+    needs: [prepare, build, build_release_image]
+    strategy:
+      matrix:
+        include: ${{ fromJson(needs.prepare.outputs.matrix_config).include }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: DeterminateSystems/nix-installer-action@main
+      - uses: docker/setup-buildx-action@v3
+      - uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Get image tag
+        id: get_version
+        run: |
+          nix run nixpkgs#nushell -- -c '
+            let version = "15"
+            let release_key = if ($version | str contains "orioledb") {
+              $"postgres($version | str replace "-" "")"
+            } else {
+              $"postgres($version)"
+            }
+            let pg_version = (open ansible/vars.yml | get postgres_release | get $release_key | str trim)
+            
+            $"pg_version=($pg_version)" | save --append $env.GITHUB_OUTPUT
+          ' 
+      - name: Merge multi-arch manifests
+        run: |
+          docker buildx imagetools create -t supabase/postgres:${{ steps.get_version.outputs.pg_version }} \
+          supabase/postgres:${{ steps.get_version.outputs.pg_version }}_amd64 \
+          supabase/postgres:${{ steps.get_version.outputs.pg_version }}_arm64
+  get_publish_version:
+    needs: [prepare, merge_manifest]
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.get_versions.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: DeterminateSystems/nix-installer-action@main
+      - name: Get version
+        id: get_versions
+        run: |
+          nix run nixpkgs#nushell -- -c '
+            let original_matrix = ${{ needs.prepare.outputs.matrix_config }}
+            let versions = $original_matrix.include
+            let processed_versions = ($versions | each { |item| 
+              let version = $item.version
+              let release_key = if ($version | str contains "orioledb") {
+                $"postgresorioledb-17"
+              } else {
+                $"postgres($version)"
+              }
+              let pg_version = (open ansible/vars.yml | get postgres_release | get $release_key | str trim)
+              
+              # Include both AMD64 and ARM64 architectures
+              [{version: $pg_version, arch: "amd64"}, {version: $pg_version, arch: "arm64"}]
+            } | flatten)
+
+            let matrix = {include: $processed_versions}
+            let matrix_json = ($matrix | to json -r)
+            $"matrix=($matrix_json)" | save --append $env.GITHUB_OUTPUT
+          '
+
+  publish:
+    needs: get_publish_version
+    strategy:
+      matrix: ${{ fromJson(needs.get_publish_version.outputs.matrix) }}
```

Comment:
We don't need a matrix here because the merge manifest job should create a single tag that can be passed down to mirror workflow.

Reply:
I ended up dropping that and doing it in a completely different way, but also made sure that after merge_manifest onward we are only dealing with the single tag produced from merge_manifest